### ISSUE TITLE
Two spaces before command-line option avoid detection

### ISSUE THREAD

**Issue by Egor-Skriptunoff** [SpiderLabs/owasp-modsecurity-crs#1019]
The rule specified at OWASP ModSecurity Core Rule Set ver.3.0.2

    file:      REQUEST-932-APPLICATION-ATTACK-RCE.conf
    section:   [ Unix direct remote command execution ]
    line:      387

seems to have a security hole:
substring `jobs -x` is detectable with single space between `jobs` and `-x`,
but is NOT detectable with two spaces in-between.

This observation was made by examining the regex.
I haven't done any testing.

This problem could be solved by replacing "jobs -x"/"java" subregex
`|j[\\\\'\"]*(?:o[\\\\'\"]*b[\\\\'\"]*s[\\\\'\"]*\s[\\\\'\"]*-[\\\\'\"]*x|a[\\\\'\"]*v[\\\\'\"]*a)|`
with
`|j[\\\\'\"]*(?:o[\\\\'\"]*b[\\\\'\"]*s[\\\\'\"]*\s[\s\\\\'\"]*-[\\\\'\"]*x|a[\\\\'\"]*v[\\\\'\"]*a)|`

**Comment by lifeforms** [comment]
Thanks for the review and report! I will investigate.

The regexps are generated by `regexp-cmdline.py` from the source files such as `regexp-932100.txt`. Afterwards, they are compressed, generating the (unreadable) end result.

```
def regexp_char(char, evasion):
    char = str.replace(char, ' ', '\s')
    char = str.replace(char, '.', '\.')
    char = str.replace(char, '-', '\-')
    char = str.replace(char, '+', r'''(?:\s|<|>).*''')
```

The entry for `jobs` in the source file is `jobs+`. It seems that the `+` should be expanded to `(?:\s|<|>).*`, which should also match double spaces, so it's not immediately clear to me why the end result does not do this. I'll have to look closer soon. Thanks again!

**Comment by Egor-Skriptunoff** [comment]
It seems that problem is with entry `jobs -x`, not with `jobs`

**Comment by illman2** [comment]
Hello Egor Skriptunoff：
      I saw your words in Stack Overflow yesterday~~

**Comment by illman2** [comment]
the plugin regexp-cmdline.py is easy to use？and where is the download site？I can't find it
in https://pypi.python.org/pypi @lifeforms

**Comment by lifeforms** [comment]
Thanks for the report @Egor-Skriptunoff , this is now addressed by #1102 and the fix will be in CRS 3.1.
